File: generate_thumbnail.js
const { Img, Text, Composition, staticFile } = require('remotion');
const { render } = require('@remotion/renderer');
const fs = require('fs');

// Load the image
const imagePath = 'downloaded_images/000001.jpg';  // Adjust the path to your image

// Create a composition with the image and text overlay
const Thumbnail = () => {
  return (
    <Composition
      width={1280}
      height={720}
      fps={30}
      durationInFrames={150}  // Just a frame duration for the composition
    >
      <Img
        src={staticFile(imagePath)}
        style={{ width: '100%', height: '100%', objectFit: 'cover' }}
      />
      <Text
        text="Welcome to San Francisco!"
        fontSize={50}
        color="white"
        style={{
          position: 'absolute',
          bottom: '100px',
          left: '50%',
          transform: 'translateX(-50%)',
        }}
      />
    </Composition>
  );
};

// Render the image
async function generateThumbnail() {
  const outputPath = 'output/thumbnail_with_text.png';
  const output = await render(<Thumbnail />);

  // Save the output image
  fs.writeFileSync(outputPath, output);
  console.log('Thumbnail generated and saved to:', outputPath);
}

generateThumbnail();

File: generate_thumbnail.py

from manim import *

class ThumbnailWithText(Scene):
    def construct(self):
        # Load the image
        image = ImageMobject("downloaded_images/000001.jpg")  # Adjust the image path
        image.scale_to_fit_height(config.frame_height)  # Fit image to frame

        # Add text overlay
        text = Text(
            "Welcome to San Francisco!",
            font="Arial-Bold",
            color=WHITE,
            weight=BOLD
        ).scale(0.5)  # Adjust text size
        text.next_to(image, DOWN, buff=0.5)  # Position text below the image

        # Add the image and text to the scene
        self.add(image, text)

        # Save a static frame (optional)
        self.pause()

# Create a separate function for downloading images
from icrawler.builtin import GoogleImageCrawler

def download_images():
    # Create a GoogleImageCrawler instance
    crawler = GoogleImageCrawler(storage={'root_dir': 'downloaded_images'})
    crawler.crawl(keyword='San Francisco', max_num=10)  # Adjust keyword and max_num as needed

if __name__ == "__main__":
    # Download images
    download_images()

    # Render the thumbnail with Manim
    ThumbnailWithText().render()
